### ISSUE TITLE
2-space indentation for TypeScript templates

### ISSUE THREAD

**Issue by kennu** [awslabs/aws-cdk#761]
When you create a new AWS CDK project based on TypeScript, all TypeScript and JSON files use 4-space indentation. I believe the generally accepted de-facto standard for both is 2-space indentation, so I propose to change them to 2 spaces.

**Comment by eladb** [comment]
We have been contemplating on whether to use 4 spaces (like VSCode/TypeScript) or 2 spaces (like [the rest of the industry](https://github.com/basarat/typescript-book/blob/master/docs/styleguide/styleguide.md#spaces)).

The ball's still in the air. 

@RomainMuller @rix0rrr, what do you say? Shall we move to 2 spaces? Now's the time probably.

**Comment by rix0rrr** [comment]
I definitely prefer a little breathing room in my code, which means more whitespace, which means 4 spaces. 2 looks ugly and feels cramped to me, so I personally don't like it.

But I also recognize that this is a bikeshed issue that we can discuss for hours, and is not worth the effort.

Only... I'm worried for the PR breakage a mass change is going to introduce.

**Comment by eladb** [comment]
> Only... I'm worried for the PR breakage a mass change is going to introduce.

Yes, I agree. That's a major concern, but just going to be more and more complex. My experience is that 2 spaces work pretty well for javascript code. It takes a bit of time to get used to (like any style), but it's perfectly fine.

I am concerned by [this](https://github.com/basarat/typescript-book/blob/master/docs/styleguide/styleguide.md#spaces) - I think there's value in aligning with the industry standard around coding style, and if there's clear evidence that we are the outlier, I rather we align.

@RomainMuller  what's your take?

**Comment by RomainMuller** [comment]
While I generally prefer having more space around stuff (more space helps with dyslexia), my personal case is that following lines horizontally is more difficult than vertically, so I should be able to adjust to a smaller indent relatively easily (anyway, large indents don't save me when indentation is more than 2-3 levels deep).

So I'm in favor of aligning with the industry standard and reclaiming some horizontal space to cram code into :)

**Comment by eladb** [comment]
Okay, here we go: #790

**Comment by kennu** [comment]
Cool :) I don't want to bikeshed with pros and cons of the actual indent amount. I'm just used to all JavaScript and JSON code being 2-space, just as I'm used to all Python code being 4-space. I'm hoping that CDK generates by default whatever most people see as the de facto, so that it's easy to copy paste code and so on.
